pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -266,15 +267,20 @@ func inferenceGraphReadiness(status v1alpha1api.InferenceGraphStatus) bool {
 }
 
 func (r *InferenceGraphReconciler) SetupWithManager(mgr ctrl.Manager, deployConfig *v1beta1api.DeployConfig) error {
-	if deployConfig.DefaultDeploymentMode == string(constants.RawDeployment) {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1alpha1api.InferenceGraph{}).
-			Owns(&appsv1.Deployment{}).
-			Complete(r)
+	ksvcFound, err := utils.IsCrdAvailable(mgr.GetConfig(), "serving.knative.dev/v1", "Service")
+	if err != nil {
+		return err
+	}
+
+	ctrlBuilder := ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1api.InferenceGraph{}).
+		Owns(&appsv1.Deployment{})
+
+	if ksvcFound {
+		ctrlBuilder = ctrlBuilder.Owns(&knservingv1.Service{})
 	} else {
-		return ctrl.NewControllerManagedBy(mgr).
-			For(&v1alpha1api.InferenceGraph{}).
-			Owns(&knservingv1.Service{}).
-			Complete(r)
+		r.Log.Info("The InferenceGraph controller won't watch serving.knative.dev/v1/Service resources because the CRD is not available.")
```

Comment:
Should this be warning?

Reply:
The `r.Log` does not have a warning function. Only `Info` and `Error`.

Reply:
Oh interesting

Reply:
Would `V(0)` meaningful?
I'm not sure it will have any outcome in the logs.

Reply:
It is not clear in the documentation, but digging a little bit about it, V(0) seems to a way to add verbosity in the logs.

Reply:
Yes it's the verbosity level.

Reply:
So if you run the controller you can do something like `--gloglevel 5` to print out only logs that falls that verbosity boundary

Reply:
I reviewed this and I think we should not raise the verbosity for this one as it helps to see why the controller would not watch the involved CRD -- and it is printed only once for the pod lifetime.

So, I think it is worth to have it as a regular Info, given `r.Log` does not have a warning function.